test/pkg/environment/environment.go

```diff
@@ -24,8 +24,10 @@ var EnvironmentName = flag.String("environment-name", env.WithDefaultString("ENV
 
 type Environment struct {
 	context.Context
-	Options *Options
-	Client  client.Client
+	Options         *Options
+	Client          client.Client
+	Monitor         *ClusterMonitor
+	restartsAtStart map[string]int
```

Comment:
Assuming they are always zero causes flakier tests.  If one test causes a crash, all subsequent tests would fail.  This way we only report a failure on the test that caused the restart.

Reply:
karpenter specific seems fine to me for now, especially if it helps maintain encapsulation.

> Assuming they are always zero causes flakier tests. If one test causes a crash, all subsequent tests would fail. This way we only report a failure on the test that caused the restart.

I'm not sure it'll be so easy to tease them apart. The first test that notices a restart should fail the entire run, as something is not okay.